### ISSUE TITLE
implement weight_norm on mps 

### ISSUE THREAD

**Issue by nwoodall** [pytorch/pytorch#104513]
### 🚀 The feature, motivation and pitch

For some models the best normalization method is using weight_norm. This is especially true when training for a hardware implementation for processing high resolution images. It's not possible to use batch_norm because the batch size is one, group/layer normalization adds calculations and latency to the hardware implementation.

### Alternatives

Allowing fallback on the CPU during training with the obvious training time penalty

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev
